metadataproxy/roles.py

```diff
@@ -155,9 +156,10 @@ def find_container(ip):
     return None
 
 
-def check_role_name_from_ip(ip, requested_role):
-    role_name = get_role_name_from_ip(ip)
-    if role_name == requested_role:
+def check_role_name_from_ip(ip, requested_role, expected_role_name=None):
```

Comment:
I think the previous logic here was ok and matches the name of the function well. Can we keep this logic and move the logging line from the route into this location?